### PR TITLE
Zip postprocessor

### PR DESCRIPTION
**Pull request by mpacer**
Create a zip postprocessor.

I feel like I did something slightly shady with how I managed the loops for the files, but I think it should work as is.

Question: by default should it clean up the files that would have otherwise been created? It feels like it kind of should…

### CODE REVIEW

**Comment by mpacer** [comment]
Starting to think that we should have a pack writer that works over byte streams in addition to the pack post-processor, that way we can use it directly and configurably from the notebook server rather than the current `require_zip` function. Also in terms of other packing formats to support I'm thinking just tar for now (maybe tar.gz too?) since that seems to cover the most common formats.

**Comment by minrk** [comment]
> by default should it clean up the files that would have otherwise been created? 

Yes, I think so.

> Also in terms of other packing formats to support I'm thinking just tar for now (maybe tar.gz too?)

I would only add tar.gz (not uncompressed .tar) for now.

File: nbconvert/postprocessors/pack.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # need to actually write this and look at the api
+    # def compression_type = DottedStringImport 
+
+    def postprocess(self, input):
+        dirname, filename = os.path.split(input)
+        base_name = os.path.splitext(filename)[0]
+        #  print(self.config.output_files_dir)
+        #  output_files_dir = (self.config.NbConvertApp.output_files_dir).format(notebook_name=base_name)
+        output_files_dir = '{notebook_name}_files'.format(notebook_name=base_name)
+        tar_filename = base_name + '.tar'
```
Comment (minrk) [inline nbconvert/postprocessors/pack.py]: When one sees `tar`, one generally means `tar.gz`, so I would assume gzip-compression in this one (`mode='w:gz'`) below. Alternately, you could have a `TarPostProcessor.compression` config trait with choices: `gz`, `bz2`, looking something like:

```python
tar_filename = base_name + '.tar.' + self.compression
...
with TarFile(tarpath, mode='w:%s' % self.compression) as tarf:
    ...
```

Comment (mpacer) [inline nbconvert/postprocessors/pack.py]: I was specifically trying to use tar in the browser as that was seeming to play nicer with javascript encodings (still not sure why) for downloading objects.

**Comment by mpacer** [comment]
in light of potentially deprecating postprocessors, i'm going to close this.

**Pull request closed without merging**